### PR TITLE
fix: product published date displaying current date in local language

### PR DESCRIPTION
**Pull request by shohag121**
When we visit the product page in the local language the product created date displays the current date instead of the actual published date.

### CODE REVIEW

**Review by nurul-umbhiya — APPROVED** [review]

File: templates/products/products-listing-row.php [hunk 1]
```diff
@@ -139,13 +139,13 @@
...
             } else {
-                $h_time = mysql2date( __( 'Y/m/d', 'dokan-lite' ), $m_time );
+                $h_time = $m_time;
             }
         }
 
         $post_date_column_time = apply_filters( 'post_date_column_time', dokan_date_time_format( $h_time, true ), $post, 'date', 'all' );
 
-        echo '<abbr title="' . esc_attr( dokan_date_time_format( $t_time ) ) . '">' . esc_html( $post_date_column_time ) . '</abbr>';
+        echo '<abbr title="' . esc_attr( dokan_date_time_format( $h_time ) ) . '">' . esc_html( $post_date_column_time ) . '</abbr>';
```
Comment (nurul-umbhiya) [inline templates/products/products-listing-row.php:148]: you can't use dokan_date_time_format() here. coz $h_time can be human readable time
echo only $h_time here

File: templates/products/products-listing-row.php [hunk 2]
```diff
@@ -139,13 +139,13 @@
             if ( $time_diff > 0 && $time_diff < 24 * 60 * 60 ) {
                 $h_time = sprintf( __( '%s ago', 'dokan-lite' ), human_time_diff( $time ) );
             } else {
-                $h_time = mysql2date( __( 'Y/m/d', 'dokan-lite' ), $m_time );
+                $h_time = $m_time;
```
Comment (nurul-umbhiya) [inline templates/products/products-listing-row.php:142]: here: use dokan_date_time_format($m_time)

Comment (nurul-umbhiya) [reply templates/products/products-listing-row.php:142]: On line 134 use $m_time = $post->post_date_gmt;

Comment (nurul-umbhiya) [reply templates/products/products-listing-row.php:142]: on second thought On line 134 use $m_time = $post->post_date; is enough, I've checked dokan_date_time_format(), this doesn't double convert time.

File: templates/products/products-listing-row.php (hunk 1 above)
Comment (shohag121) [reply templates/products/products-listing-row.php]: Code changed as per your instruction.

File: templates/products/products-listing-row.php (hunk 2 above)
Comment (shohag121) [reply templates/products/products-listing-row.php]: Bhai, I've refactored the whole block for better understanding and used `get_post_time( 'c', true, $post );` for formating.

**Comment by shohag121** [comment]
@FaraziF Bhai, Please test now.

**Comment by nurul-umbhiya** [comment]
post_date = 2021-04-28 10:14:37
post_date_gmt = 2021-04-28 04:14:37
Site timezone: UTC+6

    
$post_timestamp_GMT = 1619583277
$post_datetime_C = 2021-04-28T04:14:37+00:00
dokan_date_time_format( $post_datetime_C ) = April 28, 2021 4h14
apply_filters( 'post_date_column_time', dokan_date_time_format( $post_datetime_C, true ), $post, 'date', 'all' ) = April 28, 2021

Bottom line: Time display output should always display time, based on server timezone, this might seems wrong as I've added a new post 7pm my time but in frontend I'm seeing 6 hours before my posting time !!

@shohag121 @FaraziF

**Comment by shohag121** [comment]
@FaraziF , @nurul-umbhiya Bhaya, `get_date_from_gmt()` is used for converting time in local time format. 

I think the issue is finally fixed.

File: templates/products/products-listing-row.php [hunk 3]
```diff
@@ -126,26 +126,25 @@
...
     <td class="post-date" data-title="<?php esc_attr_e( 'Date', 'dokan-lite' ); ?>">
         <?php
+
+
+
         if ( '0000-00-00 00:00:00' == $post->post_date ) {
-            $t_time    = $h_time    = __( 'Unpublished', 'dokan-lite' );
+            $post_published_date = $human_readable_time = __( 'Unpublished', 'dokan-lite' );
             $time_diff = 0;
         } else {
```
Comment (nurul-umbhiya) [inline templates/products/products-listing-row.php:135]: a better approach without all the complexities:
```
// get current time
    $current_time = dokan_current_datetime();
    // set timezone to gmt
    $gmt_time = $current_time->setTimezone( new DateTimeZone( 'UTC' ) );
    // read post gmt time
    $post_time_gmt = $gmt_time->modify( $post->post_date_gmt );

    // get post time in wp timezone
    $post_published_ts = $post_time_gmt->setTimezone( dokan_wp_timezone() )->getTimestamp();
    $format = apply_filters( 'dokan_date_time_format', wc_date_format() . ' ' . wc_time_format() );
    // currently dokan_format_date doesn't support time format, will update this soon, 
    // right now we need to send $format from our end.
    $post_published_date = dokan_format_date( $post_published_ts, $format );

    // get human readable time
    $time_diff = $current_time->getTimestamp() - $post_time_gmt->getTimestamp();
    if ( $time_diff > 0 && $time_diff < 24 * 60 * 60 ) {
        $human_readable_time = sprintf( __( '%s ago', 'dokan-lite' ), human_time_diff( $post_time_gmt->getTimestamp() ) );
    }
```

Comment (shohag121) [reply templates/products/products-listing-row.php:135]: Refactored the code block with your code above.

Comment (nurul-umbhiya) [reply templates/products/products-listing-row.php:135]: @shohag121 thanks, @FaraziF bhai can you please test this or one more time ?

**Pull request merged**